akka-docs/src/main/paradox/typed/interaction-patterns.md

```diff
@@ -10,44 +10,86 @@ The fundamental way to interact with an actor is through @scala["tell", which is
 
- * Subscribing to an actor that will send many response messages (of the same protocol) back
- * When communicating between a parent and its children, where the protocol can be made include the messages for the interaction 
- * ???
+Java
+:  @@snip [InteractionPatternsTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/InteractionPatternsTest.java) { #request-response-protocol }
+
+
+The sender would use its own @scala[`ActorRef[Response]`]@java[`ActorRef<Response>`], which it can access through @scala[`ActorContext.self`]@java[`ActorContext.getSelf()`], for the `respondTo`. 
+
+Scala
+:  @@snip [InteractionPatternsSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/InteractionPatternsSpec.scala) { #request-response-send }
+
+Java
+:  @@snip [InteractionPatternsTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/InteractionPatternsTest.java) { #request-response-send }
+
+
+On the receiving side the `ActorRef` can then be used to send one or more responses back:
+
+Scala
+:  @@snip [InteractionPatternsSpec.scala]($akka$/akka-actor-typed-tests/src/test/scala/docs/akka/typed/InteractionPatternsSpec.scala) { #request-response-respond }
+
+Java
+:  @@snip [InteractionPatternsTest.java]($akka$/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/InteractionPatternsTest.java) { #request-response-respond }
+
+
+**Useful when:**
 
+ * Subscribing to an actor that will send many response messages back
+ * Translating between different actor message protocols
+ 
 **Problems:**
 
- * Often the response that the other actor wants to send back is not a part of the sending actor's protocol (see adapted request response or ask)
- * It is hard to detect and that a message request was not delivered or processed (see ask)
- * Unless the protocol already includes a way to provide context, for example a request id that is also sent in the response, it is not possible to tie an interaction to some specific context without introducing a new, separate, actor
+ * Actors seldom have a response message from another actor as a part of their protocol (see adapted response)
+ * It is hard to detect that a message request was not delivered or processed (see ask)
+ * Only one adaption can be made per response message type, if a new one is registered the old one is replaced,
+   for example different target actors can't have different adaption if they use the same response types, unless some
+   correlation is encoded in the messages
+ * Unless the protocol already includes a way to provide context, for example a request id that is also sent in the
+   response, it is not possible to tie an interaction to some specific context without introducing a new,
+   separate, actor
+
 
 ## Adapted Response
 
-Very often the receiving actor does not, and should, know of the protocol of the sending actor, and
+Most often the receiving actor does not, and should, know of the protocol of the sending actor, and
```

Comment:
'and should not'?

Reply:
Quite a lot of comma's, perhaps put 'and should not' between parentheses instead?

Reply:
I agree, I'll try to restructure the sentence.